src/entity/CourseSingleton.java

```diff
@@ -0,0 +1,68 @@
+package entity;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CourseSingleton {
```

Comment:
acesta se doreste a fi serviciul singleton de scriere + citire pentru un curs? 
1. nu este singleton pentru ca ii lipseste constructorul private
2. ar trebui redenumite clasele in CourseReadService/CourseWriteService si sa ai pastrate separate metodele de read/write, cerinta era sa aveti 2 servicii singleton generice, unul pentru scriere, unul pentru citire. Daca reuseai sa le faci generice, nu iti trebuiau clase separate pentru fiecare entitate in parte.